lib/view_model/contact_list/contact_list_view_model.dart

```diff
@@ -99,18 +99,22 @@ abstract class ContactListViewModelBase with Store {
 
   Future<void> delete(ContactRecord contact) async => contact.original.delete();
 
-  @computed
-  List<ContactRecord> get contactsToShow =>
-      contacts.where((element) => _isValidForCurrency(element)).toList();
+  ObservableList<ContactRecord> get contactsToShow => _currency != null ?
+  ObservableList.of(contacts.where((element) => _isValidForCurrency(element))) : contacts;
 
   @computed
-  List<WalletContact> get walletContactsToShow =>
-      walletContacts.where((element) => _isValidForCurrency(element)).toList();
+  List<WalletContact> get walletContactsToShow => _currency != null ?
+  walletContacts.where((element) => _isValidForCurrency(element)).toList() : walletContacts;
 
   bool _isValidForCurrency(ContactBase element) {
-    return _currency == null ||
-        element.type == _currency ||
-        element.type.title == _currency!.tag ||
-        element.type.tag == _currency!.tag;
+    if (element.name.contains('(Silent Payments)')) return false;
+    if (element.name.contains('MWEB)')) return false;
+
+    return element.type == _currency ||
```

Comment:
why not just keep `_currency == null ||` before this, and you can remove the checks you have in `get walletContactsToShow` and `get contactsToShow`

Reply:
moved _currency == null to  _isValidForCurrency. This check acts as a flag for the exchange screen, if it’s true, we don’t need validated contacts at all. So, it works this way as well.

Reply:
yeah I agree, but just a bit cleaner, since the loop isn't a big one anyway so it won't matter